### PR TITLE
Prevent iOS <4 crash and add headers/bridges

### PR DESCRIPTION
**Pull request by uroboro**

### CODE REVIEW

File: logging.mm [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include "logging.h"
 
 #include <objc/runtime.h>
+#include <CoreGraphics/CGAffineTransform.h>
```
Comment (DavidGoldman) [inline logging.mm]: Nit: These should probably be `#import <CoreGraphics/CoreGraphics.h>` and `#import <UIKit/UIKit.h>`

File: blocks.mm [hunk 2]
```diff
@@ -26,7 +27,7 @@ void logBlock(FILE *file, id block) {
   if (signature) {
     NSMethodSignature *methodSignature = [NSMethodSignature signatureWithObjCTypes:signature];
     Class kind = object_getClass(block);
-    fprintf(file, "<%s@%p signature=\"%s ; retType=%s", class_getName(kind), (void *)block, signature, methodSignature.methodReturnType);
+    fprintf(file, "<%s@%p signature=\"%s ; retType=%s", class_getName(kind), (__bridge void *)block, signature, methodSignature.methodReturnType);
```
Comment (DavidGoldman) [inline blocks.mm]: Does this work properly in ARC and non-ARC environments?

Comment (uroboro) [reply blocks.mm]: Yes, this satisfies Xcode with ARC enabled and Theos without it.

Comment (DavidGoldman) [reply blocks.mm]: Gotcha, thanks

File: logging.mm (hunk 1 above)
Comment (uroboro) [reply logging.mm]: Fixed!

**Pull request merged**